filebeat/harvester/reader.go

```diff
@@ -71,6 +74,12 @@ func (r *logFileReader) Read(buf []byte) (int, error) {
 	}
 
 	for {
+		select {
+		case <-r.done:
+			return 0, nil
+		default:
+		}
+
 		n, err := r.fs.Read(buf)
```

Comment:
what if we block on Read? Consider closing the file and do check on r.done being closed in EOF checks